### PR TITLE
Do not report exception message in fault details

### PR DESCRIPTION
**Pull request by dibarbet**
First part of fixes for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2032081

### CODE REVIEW

File: src/VisualStudio/Core/Def/Watson/FaultReporter.cs [hunk 1]
```diff
@@ -252,8 +252,9 @@ private static string GetDescription(Exception exception)
         {
         }
 
-        // If we couldn't get a stack, do this
-        return exception.Message;
+        // If we couldn't get a stack, report a generic message.
+        // The exception message is already reported in a separate cred-scanned property.
```
Comment (dibarbet) [inline src/VisualStudio/Core/Def/Watson/FaultReporter.cs:256]: the fault description field is not client side cred-scanned by VS telemetry.  We should not report the message again here, there is already an exception message property that is cred-scanned.

File: src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.cs [hunk 2]
```diff
@@ -78,10 +78,9 @@ internal static bool ResetRenameTrackingStateWorker(Workspace workspace, Documen
                 if (textBuffer == null)
                 {
                     var ex = new InvalidOperationException(string.Format(
-                        "document with name {0} is open but textBuffer is null. Textcontainer is of type {1}. SourceText is: {2}",
+                        "document with name {0} is open but textBuffer is null. Textcontainer is of type {1}.",
```
Comment (dibarbet) [inline src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.cs:81]: This doesn't make much of a difference with the below change (and assuming the message field is cred scanned), but in general we shouldn't be reporting the full source in fault telemetry.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by arkalyanms — APPROVED** [review]

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**